### PR TITLE
add manifests for initial deployment of sippy

### PR DESCRIPTION
**Pull request by deads2k**
builds on https://github.com/kubernetes/k8s.io/pull/1614

This adds manifests to deploy and run sippy.  This is similar to what I have in my private toy cluster, but I need to do a little bit of alignment and work out ingresses instead of routes.

/assign @wojtek-t 

/hold

explicit hold until I'm able to test these locally.

### CODE REVIEW

File: groups/groups.yaml
Comment (wojtek-t) [inline groups/groups.yaml:244]: nit: This is already merged - please rebase

File: wg-reliability/README.md
Comment (wojtek-t) [inline wg-reliability/README.md:6]: We should probably request public IP and DNS now (given we already have support for running it).

I'm wondering if we want "sippy.k8s.io" or something more generic?

File: wg-reliability/sippy/service.yaml
Comment (wojtek-t) [inline wg-reliability/sippy/service.yaml:2]: We also want Ingress and Certificate too, as e.g. here:
https://github.com/kubernetes/k8s.io/tree/master/perf-dash.k8s.io

File: wg-reliability/sippy/deployment.yaml
Comment (wojtek-t) [inline wg-reliability/sippy/deployment.yaml:39]: Do we want both? Or maybe we want to start just with blocking?

Or actually, I guess it's probably moot anyway, because we probably want to create our own dashboard for this purpose...

File: wg-reliability/sippy/deployment.yaml
Comment (wojtek-t) [inline wg-reliability/sippy/deployment.yaml:53]: What is this? I didn't see that anywhere.
I think you can simply run:
./sippy --fetch-data /usr/local/google/home/wojtekt/sippy_data --dashboard=kube-master=sig-release-master-blocking=

Or I guess, you might wanted this to run periodically already.
Do we want a container? Or do we want a CronJob that will be doing this?

File: wg-reliability/sippy/deployment.yaml
Comment (wojtek-t) [inline wg-reliability/sippy/deployment.yaml:58]: Isn't 30 default?

File: wg-reliability/sippy/pvc.yaml
Comment (wojtek-t) [inline wg-reliability/sippy/pvc.yaml:11]: I guess we want to set some StorageClass, right?

File: wg-reliability/sippy/deployment.yaml
Comment (wojtek-t) [reply wg-reliability/sippy/deployment.yaml:53]: Ahh - I guess with CronJob, we may have a problem with mounting for different pods...

So we probably want this to be a container within a pod.

**Review by spiffxp — CHANGES_REQUESTED** [review]
I would rather see these manifests in a folder called `sippy` or `sippy.k8s.io`, in a `sippy` namespace to match.

If we're going down the path of assigning group ownership to apps deployed in cluster (we should! accountability is great!) I would rather see a `sig: foo` label, given that wg's don't own code.

I would suggest `sig: architecture` or `sig: release` to start... though long term it should be `sig: testing` when we have bandwidth.

File: sippy/sippy/pvc.yaml
Comment (wojtek-t) [inline sippy/sippy/pvc.yaml:12]: @spiffxp - what storage class we have in the community cluster?

**Review by wojtek-t — COMMENTED** [review]
This looks fine to me (modulo missing storage class).

File: sippy/sippy/pvc.yaml
Comment (ameukam) [reply sippy/sippy/pvc.yaml:12]: @wojtek-t we have 2 storage classes in the cluster : 

```shell
kubectl get sc
NAME                 PROVISIONER            AGE
ssd                  kubernetes.io/gce-pd   334d
standard (default)   kubernetes.io/gce-pd   551d
```

Comment (wojtek-t) [reply sippy/sippy/pvc.yaml:12]: Thanks!

@deads2k - standard should be good enough for us;
 Though I would visibly increase the size, say to 10GB or sth.

File: groups/groups.yaml
Comment (spiffxp) [inline groups/groups.yaml:598]: You'll probably need to rebase, I did this in https://github.com/kubernetes/k8s.io/pull/1844

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/hold
Remove /hold when ready to deploy

Thanks for changing the namespace, I ended up setting it up via https://github.com/kubernetes/k8s.io/pull/1844 so you'll likely need to rebase

File: sippy/sippy/ingress.yaml
Comment (spiffxp) [inline sippy/sippy/ingress.yaml:1]: I'm not sure if aaa is at a version that supports this

**Review by spiffxp — APPROVED** [review]
/lgtm

File: sippy/sippy/pvc.yaml
Comment (wojtek-t) [reply sippy/sippy/pvc.yaml:12]: @deads2k - friendly ping

File: sippy/sippy/ingress.yaml
Comment (wojtek-t) [reply sippy/sippy/ingress.yaml:1]: good catch - the cluster is still in 1.16 (will be upgraded to 1.17 soon, but ingress/v1 is only in 1.19), so we should use v1beta1 stil...

**Comment by deads2k** [comment]
/hold

We will merge on monday when everyone is back in the office and ready for potential excitement

**Comment by deads2k** [comment]
/hold cancel

here we go!

**Comment by sbueringer** [comment]
@deads2k for even more excitement. Is there a specific issue where I can follow the implementation?

**Comment by sbueringer** [comment]
Ah sorry that was misleading. I meant more like the deployment of sippy for the kubernetes repos. I saw the DNS name `sippy.k8s.io`, but it's not available yet. So I assumed there might be an issue somewhere (which I can subscribe to) to stay up-to-date about the deployment implementation.

**Comment by deads2k** [comment]
> Ah sorry that was misleading. I meant more like the deployment of sippy for the kubernetes repos. I saw the DNS name `sippy.k8s.io`, but it's not available yet. So I assumed there might be an issue somewhere (which I can subscribe to) to stay up-to-date about the deployment implementation.

As it happens, I was just starting to look now and I haven't worked out where my resources went.

```
deads@cloudshell:~$ kubectl --context=prod-aaa -nsippy get pod
No resources found in sippy namespace.
deads@cloudshell:~$ kubectl --context=prod-aaa -nsippy get deployment
No resources found in sippy namespace.
deads@cloudshell:~$ kubectl --context=prod-aaa -nsippy get all
Error from server (Forbidden): replicationcontrollers is forbidden: User "deads@redhat.com" cannot list resource "replicationcontrollers" in API group "" in the namespace "sippy": r
equires one of ["container.replicationControllers.list"] permission(s).
Error from server (Forbidden): daemonsets.apps is forbidden: User "deads@redhat.com" cannot list resource "daemonsets" in API group "apps" in the namespace "sippy": requires one of 
["container.daemonSets.list"] permission(s).
deads@cloudshell:~$ kubectl --context=prod-aaa -nsippy get ingress
No resources found in sippy namespace.
```

**Comment by sbueringer** [comment]
Ah got it :). I'll just follow the update here then.

**Comment by ameukam** [comment]
@deads2k 👋🏾 , resources deployment is still a manual action in `aaa`.

**Comment by spiffxp** [comment]
Part of https://github.com/kubernetes/k8s.io/issues/1900

**Pull request merged**
